flink-libraries/flink-state-processing-api/src/main/java/org/apache/flink/state/api/output/SnapshotUtils.java

```diff
@@ -52,12 +53,13 @@ private SnapshotUtils() {}
             boolean isExactlyOnceMode,
             boolean isUnalignedCheckpoint,
             Configuration configuration,
-            Path savepointPath)
+            Path savepointPath,
+            SnapshotType snapshotType)
```

Comment:
Generally, I think we should limit the snapshot type as full snapshot or savepoint instead of any kind of snapshot type.

Reply:
Good catch, I will add the limit and the unit test. THX!

Reply:
After re-think current stats in Flink, I currently prefer we could only support native and canonical savepoint. You can try to use native savepoint to see whether it could meet your previous requestls. The native savepoint is much close to full checkpoint and it could also support relocating the checkpoint data to other place without modifying the metadata.